sql/src/main/java/org/apache/druid/sql/calcite/aggregation/builtin/CountSqlAggregator.java

```diff
@@ -134,15 +134,17 @@ public Aggregation toDruidAggregation(
     } else {
       // Not COUNT(*), not distinct
       // COUNT(x) should count all non-null values of x.
-      return Aggregation.create(createCountAggregatorFactory(
+      AggregatorFactory theCount = createCountAggregatorFactory(
```

Comment:
How about making createCountAggregatorFactory return Aggregation instead of AggregatorFactory? It seems like that would be less error-prone. (Currently, since it returns an AggregatorFactory that might reference some virtual columns, all callers must check for that, which is easy to forget [hence this bug] and makes the callers more complex.)

Reply:
Instead I have just removed the `VirtualColumn` tracking from `Aggregation` entirely in favor of replacing it with a new `getRequiredColumns ` that can list all columns between agg factories and the post agg, and using the new find method on the `VirtualColumnRegistry` to look them up.